pkg/datasource/accessors.go

```diff
@@ -379,6 +388,37 @@ func (a *fieldAccessor) Flags() uint32 {
 	return a.f.Flags
 }
 
+func (a *fieldAccessor) Tags() []string {
+	return slices.Clone(a.f.Tags)
+}
+
+func (a *fieldAccessor) HasAllTagsOf(tags ...string) bool {
+	for _, t1 := range tags {
+		found := false
+		for _, t2 := range a.f.Tags {
+			if t1 == t2 {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
They are not sorted - we could enforce that, though. I think it's a good idea to do it (because I also plan to use hashes to compare gadgetInfo structs later on, for which they should be sorted as well), but I'll postpone that for now.